selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/preferences/ContributeTabTest.java

```diff
@@ -0,0 +1,288 @@
+        askDialog.confirmAndWaitClosed();
+      }
+
+      if (preferences.isPreferencesFormOpened()) {
+        preferences.clickOnCloseButton();
+      }
+
+      preferences.waitPreferencesFormIsClosed();
+
+    } catch (Exception e) {
+      LOG.error(e.getLocalizedMessage(), e);
+    }
+  }
+
+  @Test
+  public void checkSettingContributeTab() {
+    firstProjectUrl = testRepo.getHtmlUrl() + ".git";
+    secondProjectUrl = testRepo2.getHtmlUrl() + ".git";
+
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(THIRD_PROJECT_NAME);
+
+    // import the first and second projects
+    git.importJavaApp(firstProjectUrl, FIRST_PROJECT_NAME, MAVEN);
+    git.importJavaApp(secondProjectUrl, SECOND_PROJECT_NAME, MAVEN);
+
+    // check opening the PR panel by select a project
+    pullRequestPanel.waitClosePanel();
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+
+    // check the text when a project is not under VCS
+    projectExplorer.waitAndSelectItem(THIRD_PROJECT_NAME);
+    pullRequestPanel.waitTetxNotVcsProject(EXP_TEXT_NOT_VCS);
+
+    // check the 'Contribute' checkbox is true by default
+    openContributeTab();
+    preferences.waitContributeCheckboxIsSelected();
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkRefreshAndSaveButton() {
+    openContributeTab();
+
+    preferences.setAndWaitStateContributeCheckbox(true);
+    preferences.clickOnOkBtn();
+
+    assertFalse(preferences.saveButtonIsEnabled());
+
+    // check the 'Refresh button
+    preferences.clickOnContributeCheckbox();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    assertTrue(preferences.saveButtonIsEnabled());
+
+    preferences.clickRefreshButton();
+    preferences.waitContributeCheckboxIsSelected();
+
+    assertFalse(
+        preferences.saveButtonIsEnabled(),
+        "Known issue https://github.com/eclipse/che/issues/9959");
+
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkSwitchProjectsWhenContributeIsFalse() {
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(true);
+
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(false);
+
+    // switch between projects
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitRepoUrl(firstProjectUrl);
+    pullRequestPanel.waitProjectName(FIRST_PROJECT_NAME);
+
+    projectExplorer.waitAndSelectItem(SECOND_PROJECT_NAME);
+    pullRequestPanel.waitRepoUrl(secondProjectUrl);
+    pullRequestPanel.waitProjectName(SECOND_PROJECT_NAME);
+
+    projectExplorer.waitAndSelectItem(THIRD_PROJECT_NAME);
+    pullRequestPanel.waitTetxNotVcsProject(EXP_TEXT_NOT_VCS);
+  }
+
+  @Test(priority = 1)
+  public void checkAutomaticChangeContributeToFalse() {
```

Comment:
What does "Contribute to false" mean?

Reply:
Preference property value is changed automatically to "false":

* user clicks "Hide" button of the Tooling Part Stack panel
* user hit "Options --> Hide" action of the Tooling Part Stack panel
* user hit action from menu Assistant --> Tool Windows --> Contribute when Contribute Part is active

To inform user how change preference property notification is displayed:

` To activate Contribute Panel by clicking on a project go to Profile -> Preferences -> Git -> Contribute `